cloudformation/stack_topic_handler.js

```diff
@@ -205,20 +315,44 @@ async function handleASGMessage(message, context) {
       const check = healthChecks[i];
 
       // Remove all health checks if only one IP or if a health check is registered to an IP not used.
-      if (ipAddresses.length === 1 || !ipAddresses.find(ip => check.HealthCheckConfig.IPAddress === ip)) {
-        //console.log("deleting check " + check.Id);
+      if (
+        ipAddresses.length === 1 ||
+        !ipAddresses.find((ip) => check.HealthCheckConfig.IPAddress === ip)
+      ) {
+        console.log("deleting health check " + check.Id);
 
         try {
-          await promisify(route53.deleteHealthCheck.bind(route53))({ HealthCheckId: check.Id });
-        } catch(e) {}
+          await promisify(route53.deleteHealthCheck.bind(route53))({
+            HealthCheckId: check.Id,
+          });
+        } catch (e) {}
       }
     }
   } else {
-    console.log("Unsupported ASG event: " + asgName + " " + asgEvent);
     context.done("Unsupported ASG event: " + asgName + " " + asgEvent);
   }
 }
 
+/**
+ * Takes a recordName and checks it against the assigned hostname adjectives/nouns
+ * returns boolean whether the recordName is an assigned hostname
+ */
+function isAssignedHostName(recordName, domainUrl) {
+  if (!recordName) return false;
+
+  const splitRecordName = recordName.split("-");
+  if (splitRecordName.length >= 2) {
+    // ["adjective","noun" || "noun.url.com", ?"local.url.com"]
+    const adjective = splitRecordName[0];
+    const noun = splitRecordName[1].includes("." + domainUrl)
```

Comment:
This doesn't work if the domain contains hyphens. For example, "focused-hobbit.hubs-cloud-test-internal.net."